src/ubx-message.toit

```diff
@@ -200,90 +503,833 @@ class AckAck extends Message:
+  static ID ::= 0x04
+
+  /**
+  Constructs a reset message.
+
+  The default parameters are a controlled software reset with a cold start.
+  See the description for other parameter options.
+  */
+  constructor --clear-sections=0xFFFF --reset-mode=2:
+    super.private_ Message.CFG ID (ByteArray 4)
+    LITTLE-ENDIAN.put-uint16 payload 0 clear-sections
+    LITTLE-ENDIAN.put-uint8  payload 2 reset-mode
+    LITTLE-ENDIAN.put-uint8  payload 3 Message.RESERVED_
+
+  id-string_ -> string:
+    return "RST"
+
+/**
+The UBX-NAV-STATUS message.
+
+The receiver navigation status.
+*/
+// https://www.u-blox.com/en/docs/UBX-13003221#%5B%7B%22num%22%3A1204%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C0%2C559.1%2Cnull%5D
+class NavStatus extends Message:
+  /** The UBX-NAV-STATUS message ID. */
+  static ID ::= 0x03
+
+  /** Unknown GNSS fix. */
+  static NO-FIX ::= 0
+  /** Dead reckoning only. */
+  static DEAD-RECKONING-ONLY ::= 1
+  /** 2D fix. */
+  static FIX-2D ::= 2
+  /** 3D fix. */
+  static FIX-3D ::= 3
+  /** GPS and dead reckoning. */
+  static GPS-DEAD-FIX ::= 4
+  /** Time only fix. */
+  static TIME-ONLY ::= 5
+  /** Constructs a poll UBX-NAV-STATUS message. */
+
+  static PACK-FIX-TYPES ::= {
+    NO-FIX: "NO-FIX",
+    DEAD-RECKONING-ONLY : "DEAD-RECKONING-ONLY",
+    FIX-2D : "FIX-2D",
+    FIX-3D : "FIX-3D",
+    GPS-DEAD-FIX : "GPS-DEAD-FIX",
+    TIME-ONLY : "TIME-ONLY"}
+
+  constructor.poll:
+    super.private_ Message.NAV ID #[]
+
+  constructor.private_ payload:
+    super.private_ Message.NAV ID payload
+
+  id-string_ -> string:
+    return "STATUS"
+
+  /** The GPS interval time of week of the navigation epoch. */
+  itow -> int:
+    assert: not payload.is-empty
+    return LITTLE-ENDIAN.uint32 payload 0
+
+  /**
+  Returns the type of fix.
+
+  One of $NO-FIX, $DEAD-RECKONING-ONLY, $FIX-2D, $FIX-3D, $GPS-DEAD-FIX, $TIME-ONLY.
+  */
+  gps-fix -> int:
+    assert: not payload.is-empty
+    //return LITTLE-ENDIAN.uint8 payload 4
+    return LITTLE-ENDIAN.uint8 payload 4
+
+  // Thinking to remove this and have the user/driver do it via the PACK... static.
+  gps-fix-text -> string:
+    assert: not payload.is-empty
+    return PACK-FIX-TYPES[gps-fix]
+
+  /**
+  Navigation status flags.
+  See receiver specification for details.
+  */
+  flags -> int:
+    assert: not payload.is-empty
+    return LITTLE-ENDIAN.uint8 payload 5
+
+  /**
+  Fix status information
+  See receiver specification for details.
+  */
+  fix-stat -> int:
+    assert: not payload.is-empty
+    return LITTLE-ENDIAN.uint8 payload 6
+
+  /**
+  Additional status information.
+  See receiver specification for details.
+  */
+  flags2 -> int:
+    assert: not payload.is-empty
+    return LITTLE-ENDIAN.uint8 payload 7
+
+  /** Time to first fix in milliseconds. */
+  time-to-first-fix -> int:
+    assert: not payload.is-empty
+    return LITTLE-ENDIAN.uint32 payload 8
+
+  /** Milliseconds since startup or reset. */
+  msss -> int:
```

Comment:
Not a fan of these super shortened identifiers.
If the rest of the package uses them too, keep. Otherwise make them more readable.